ansible_base/jwt_consumer/common/auth.py

```diff
@@ -153,14 +157,22 @@ def map_user_fields(self):
             old_value = getattr(self.user, attribute, None)
             new_value = self.token.get('user_data', {}).get(attribute, None)
             if old_value != new_value:
+                if attribute == "username" and get_user_model().objects.filter(username=new_value).exists():
```

Comment:
Would it be faster to not check here and just try/catch the save below and drop this error on a duplicate key error? Although we would have to try the save a second time so maybe not? But it should be easy to recover if we used the update_fields and just droped the username on a duplicate key error.

Reply:
I think this is fine. We only trigger the second conditional if the first is false, and I don't expect usernames to change very often.